### DISCUSSION TITLE
Performance issues - CPU or Wifi?

### DISCUSSION THREAD

**Discussion by ochompsky** [tesla-android/issue-tracker#116] [General]
Hi, so i've setup everything but the performance isn't even close in the videos.  The video lags by a matter of seconds and doesn't scroll smoothly.  The in-browser audio is also very skippy/static.  

- I've overclocked the RPI using the settings in the install guide.  I am using [this case](https://www.amazon.com/gp/product/B07VWM4J4L/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1) with active cooling.  
- I've setup all the recommended AutoKit settings. 
- I've tried moving the RPI to right next to the antenna in the passenger mirror to see if wifi is the problem.  Also tried running it in an area with no other wifi connections to see if there is some interference.  
- I am on the latest version of the AutoKit APK and CarlinkKit adapter firmware.  
- SD card is 64Gb Samsung Evo Plus

Despite all of this, it's not very usable and I can't tell where the bottleneck is.   I have a RPI 4 2GB model.  Is the lack of ram the problem?  Also, running Tesla 2022.36.6.

**Comment by mikegapinski** [comment]
The 2GB raspberry Pi is not a problem at all. A few things worth checking:
- Check if the USB-C cable is not low quality, some of them cannot deliver enough power. I had no problems with Anker
- Clear the browser data in your car and reboot the system with scroll wheels while holding the break pedal
- Your case is metal, metal interferes with Wi-Fi by forming a cage around your pi, it might degrade the signal strength significantly

**Reply by ochompsky** [reply]
the problem was the metal case!  wow, im amazed at how much that reduced the quality of the experience.  i went ahead and removed the overclock, removed the case and ran a wire to keep it in the glovebox, as close to the antenna as possible.  it's MUCH better.  i'll try heatsinks for now and possibly get a plastic case for the summer months.

**Reply by mikegapinski** [reply]
Don't know what your model is, but it turns out that newer model 3s have the Wi-Fi antenna near the rear windshield in the overhead pillar. Try the center console also

**Reply by ochompsky** [reply]
i have a 2020 m3 performance.  i think its in the side mirror, performance does seem better there than center console.  i would get 1-2 bars but in the glovebox it seems to be better, although that may also just be the lack of a case now.

**Comment by mikegapinski** [comment]
The bars are inaccurate, I sometimes get 1 or 3 without moving the pi

Wysyłane z aplikacji Outlook dla systemu iOS<https://aka.ms/o0ukef>
________________________________
Od: ochompsky ***@***.***>
Wysłane: Tuesday, November 1, 2022 9:18:03 PM
Do: tesla-android/issue-tracker ***@***.***>
DW: Michał Gapiński ***@***.***>; Comment ***@***.***>
Temat: Re: [tesla-android/issue-tracker] Performance issues - CPU or Wifi? (Discussion #116)


i have a 2020 m3 performance. i think its in the side mirror, performance does seem better there than center console. i would get 1-2 bars but in the glovebox it seems to be better, although that may also just be the lack of a case now.

**Comment by paulharman** [comment]
Can we use a USB wifi stick to improve the wifi connection?

**Reply by mikegapinski** [reply]
There is no point in doing this, in intel cars the Wi-Fi is weak. They've reworked this in ryzen cars, even when using the same antena it's much faster
